tables/s3_server_access_log/s3_server_access_log.go

```diff
@@ -0,0 +1,206 @@
+package s3_server_access_log
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+)
+
+type S3ServerAccessLog struct {
+	schema.CommonFields
+
+	BucketOwner        string    `json:"bucket_owner"`
+	Bucket             string    `json:"bucket"`
+	Timestamp          time.Time `json:"timestamp"`
+	RemoteIP           string    `json:"remote_ip"`
+	Requester          string    `json:"requester"`
+	RequestID          string    `json:"request_id"`
+	Operation          string    `json:"operation"`
+	Key                *string   `json:"key,omitempty"`
+	RequestURI         *string   `json:"request_uri,omitempty"`
+	HTTPStatus         *int      `json:"http_status,omitempty"`
+	ErrorCode          *string   `json:"error_code,omitempty"`
+	BytesSent          *int64    `json:"bytes_sent,omitempty"`
+	ObjectSize         *int64    `json:"object_size,omitempty"`
+	TotalTime          *int      `json:"total_time,omitempty"`
+	TurnAroundTime     *int      `json:"turn_around_time,omitempty"`
+	Referer            *string   `json:"referer,omitempty"`
+	UserAgent          *string   `json:"user_agent,omitempty"`
+	VersionID          *string   `json:"version_id,omitempty"`
+	HostID             *string   `json:"host_id,omitempty"`
+	SignatureVersion   *string   `json:"signature_version,omitempty"`
+	CipherSuite        *string   `json:"cipher_suite,omitempty"`
+	AuthenticationType *string   `json:"authentication_type,omitempty"`
+	HostHeader         *string   `json:"host_header,omitempty"`
+	TLSVersion         *string   `json:"tls_version,omitempty"`
+	AccessPointArn     *string   `json:"access_point_arn,omitempty"`
+	AclRequired        *bool     `json:"acl_required,omitempty"`
+}
+
+func NewS3ServerAccessLog() *S3ServerAccessLog {
+	return &S3ServerAccessLog{}
+}
+
+func (l *S3ServerAccessLog) InitialiseFromMap(m map[string]string) error {
+	for key, value := range m {
+		switch key {
+		case "bucket_owner":
+			l.BucketOwner = value
+		case "bucket":
+			l.Bucket = value
+		case "timestamp":
+			ts, err := time.Parse("02/Jan/2006:15:04:05 -0700", value)
+			if err != nil {
+				return fmt.Errorf("error parsing timestamp: %w", err)
+			}
+			l.Timestamp = ts
+		case "remote_ip":
+			l.RemoteIP = value
+		case "requester":
+			l.Requester = value
+		case "request_id":
+			l.RequestID = value
+		case "operation":
+			l.Operation = value
+		case "key":
+			if value != "-" {
```

Comment:
Sometimes `-` in the logs mean different things, e.g., `-` in `requester` means unauthenticated, but for `key` it means no object was involved. I'm not sure if we should always be removing them or not based on preserving actual log data.